shared/src/utils/validators/is-id-map.ts

```diff
@@ -0,0 +1,61 @@
+import { plainToInstance, Transform } from 'class-transformer';
+import type { ValidationOptions } from 'class-validator';
+import { isUUID, ValidateNested } from 'class-validator';
+import type { Constructor } from '../constructor';
+import type { UUID } from '../uuid';
+import { combineDecorators } from './combine-decorators';
+import { createMapValidator } from './create-map-validator';
+import { isValidObject } from './is-valid-object';
+
+export function isIdMap<T extends object>(
+    type: Constructor<T>,
+    getId: (value: T) => UUID,
+    valueToBeValidated: unknown
+): boolean {
+    return createMapValidator<UUID, T>({
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        keyValidator: <(key: unknown) => key is UUID>((key) => isUUID(key, 4)),
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        valueValidator: <(value: unknown) => value is T>(
+            ((value) => isValidObject(type, value))
+        ),
+        consistencyValidator: (key, value) =>
+            getId(plainToInstance(type, value)) === key,
+    })(valueToBeValidated);
+}
```

Comment:
Can this be removed? I believe it is nowhere used.

Reply:
Ah, good point. However, as I said [here](https://github.com/hpi-sam/digital-fuesim-manv/pull/125#discussion_r988392800), it's a convention to have such a method. I don't think I could rewrite the decorator stuff to be used without a decorator. So I'll have another think about this, but I tend to just remove this, maybe add a comment to look here if you need such an implementation in the future, as we're currently not using it.